<div style="background-color: #f4f6f7; padding: 20px; border-radius: 10px; border-left: 5px solid #005a9e; font-family: sans-serif;">
    <h1 style="color: #005a9e; margin-top: 0;">Predicción de aterrizaje de la primera etapa del Falcon 9 de SpaceX</h1>
    <p><strong>Tiempo estimado necesario:</strong> 60 minutos</p>
    <p><strong>La primera etapa del Falcon 9 aterrizará con éxito</strong></p>
    <p>En este laboratorio, realizaremos un Análisis Exploratorio de Datos (EDA) para encontrar patrones en los datos y determinar cuál será la etiqueta para entrenar modelos supervisados.</p>
    <p>En el conjunto de datos, hay varios casos diferentes en los que el propulsor no aterrizó con éxito. A veces se intentó un aterrizaje pero falló debido a un accidente; por ejemplo, <em>True Ocean</em> significa que el resultado de la misión fue un aterrizaje exitoso en una región específica del océano, mientras que <em>False Ocean</em> significa que el aterrizaje no tuvo éxito en esa región. <em>True RTLS</em> significa que aterrizó con éxito en una plataforma terrestre, y <em>False RTLS</em> significa que el aterrizaje en la plataforma terrestre no tuvo éxito. <em>True ASDS</em> significa aterrizaje exitoso en un barco dron, y <em>False ASDS</em> significa aterrizaje fallido en un barco dron.</p>
    <p>En este laboratorio, convertiremos principalmente esos resultados en etiquetas de entrenamiento (Training Labels), donde <code>1</code> significa que el propulsor aterrizó con éxito y <code>0</code> significa que no tuvo éxito.</p>
    <hr style="border: 1px solid #d0d7de;">
    <h3 style="color: #005a9e;">Qué hacemos</h3>
    <p>Para iniciar, prepararemos nuestro entorno de trabajo importando las librerías base necesarias y validando que estamos ejecutando las versiones correctas recomendadas (Python 3.12 y pandas 3.x) para este laboratorio.</p>
</div>


<div style="background-color: #f4f6f7; padding: 15px; border-left: 5px solid #0f62fe;">
    <h2 style="color: #0f62fe; margin-top: 0;">Importar Librerías y Definir Funciones Auxiliares</h2>
    <p><strong>Qué hacemos:</strong> Importamos las librerías fundamentales para la manipulación y análisis de datos (pandas) y operaciones matemáticas sobre arreglos (NumPy). Además, incluimos el bloque de diagnóstico recomendado para validar que estamos ejecutando el entorno <code>decision-tools-312</code> con Python 3.12 y pandas 3.0.3.</p>
    <hr style="border: 1px solid #d1d5db;">

In [1]:
import sys
import warnings

# Pandas para manipulación y análisis de datos
import pandas as pd
# NumPy para arreglos multidimensionales y funciones matemáticas
import numpy as np

warnings.filterwarnings("ignore")

print("Python:", sys.version)
print("pandas:", pd.__version__)
print("numpy:", np.__version__)

Python: 3.12.13 | packaged by conda-forge | (main, Mar  5 2026, 16:36:12) [MSC v.1944 64 bit (AMD64)]
pandas: 3.0.3
numpy: 2.5.2


<div style="background-color: #f4f6f7; padding: 15px; border-left: 5px solid #0f62fe;">
    <h2 style="color: #0f62fe; margin-top: 0;">Análisis de Datos: Cargar el Dataset</h2>
    <p><strong>Qué hacemos:</strong> Cargamos el conjunto de datos de SpaceX desde la URL provista utilizando <code>pandas.read_csv()</code> y solicitamos las primeras 10 filas para inspeccionar su estructura inicial y el tipo de variables que contiene. Separar la URL en una variable hace que el código sea más legible.</p>
    <hr style="border: 1px solid #d1d5db;">
    <p><strong>Qué observar:</strong> Deberías ver una tabla renderizada por pandas con 10 filas (índices del 0 al 9). Revisa brevemente los nombres de las columnas para familiarizarte con las dimensiones del DataFrame.</p>
</div>

In [2]:
# Definir la URL del dataset de SpaceX
url = "https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBM-DS0321EN-SkillsNetwork/datasets/dataset_part_1.csv"

# Cargar los datos en un DataFrame
df = pd.read_csv(url)

# Visualizar las primeras 10 filas
df.head(10)

,FlightNumber,Date,BoosterVersion,PayloadMass,Orbit,LaunchSite,Outcome,Flights,GridFins,Reused,Legs,LandingPad,Block,ReusedCount,Serial,Longitude,Latitude
0,1,2010-06-04,Falcon 9,6104.959412,LEO,CCAFS SLC 40,None None,1,False,False,False,NaN,1.0,0,B0003,-80.577366,28.561857
1,2,2012-05-22,Falcon 9,525.000000,LEO,CCAFS SLC 40,None None,1,False,False,False,NaN,1.0,0,B0005,-80.577366,28.561857
2,3,2013-03-01,Falcon 9,677.000000,ISS,CCAFS SLC 40,None None,1,False,False,False,NaN,1.0,0,B0007,-80.577366,28.561857
3,4,2013-09-29,Falcon 9,500.000000,PO,VAFB SLC 4E,False Ocean,1,False,False,False,NaN,1.0,0,B1003,-120.610829,34.632093
4,5,2013-12-03,Falcon 9,3170.000000,GTO,CCAFS SLC 40,None None,1,False,False,False,NaN,1.0,0,B1004,-80.577366,28.561857
5,6,2014-01-06,Falcon 9,3325.000000,GTO,CCAFS SLC 40,None None,1,False,False,False,NaN,1.0,0,B1005,-80.577366,28.561857
6,7,2014-04-18,Falcon 9,2296.000000,ISS,CCAFS SLC 40,True Ocean,1,False,False,True,NaN,1.0,0,B1006,-80.577366,28.561857
7,8,2014-07-14,Falcon 9,1316.000000,LEO,CCAFS SLC 40,True Ocean,1,False,False,True,NaN,1.0,0,B1007,-80.577366,28.561857
8,9,2014-08-05,Falcon 9,4535.000000,GTO,CCAFS SLC 40,None None,1,False,False,False,NaN,1.0,0,B1008,-80.577366,28.561857
9,10,2014-09-07,Falcon 9,4428.000000,GTO,CCAFS SLC 40,None None,1,False,False,False,NaN,1.0,0,B1011,-80.577366,28.561857


<div style="background-color: #f4f6f7; padding: 15px; border-left: 5px solid #24a148;">
    <h2 style="color: #24a148; margin-top: 0;">Interpretación de los Resultados</h2>
    <p><strong>Qué significa esto:</strong> Has cargado el dataset correctamente en tu entorno pandas 3.0.3. Al examinar estas primeras 10 filas, confirmamos la estructura tabular de los datos de SpaceX. Tenemos variables categóricas importantes como <code>Orbit</code>, <code>LaunchSite</code> y <code>Outcome</code>, junto con variables numéricas como <code>PayloadMass</code>.</p>
    <p>Un detalle clave a observar es la presencia de valores <code>NaN</code> en columnas como <code>LandingPad</code>. Esto nos indica que el dataset tiene datos faltantes, lo cual es habitual en escenarios reales y sugiere que los siguientes pasos del laboratorio probablemente se enfocarán en la exploración (Exploratory Data Analysis) y limpieza (Data Wrangling) de estas variables.</p>
    <hr style="border: 1px solid #d1d5db;">

<div style="background-color: #f4f6f7; padding: 15px; border-left: 5px solid #0f62fe;">
    <h2 style="color: #0f62fe; margin-top: 0;">Análisis Exploratorio: Faltantes y Tipos de Datos</h2>
    <p><strong>Qué hacemos:</strong> Ejecutamos dos pasos fundamentales del análisis exploratorio (EDA). Primero, calculamos el porcentaje exacto de valores nulos o faltantes en cada columna para saber qué atributos requieren limpieza o imputación. Segundo, identificamos los tipos de datos de cada variable para distinguir entre columnas numéricas y categóricas.</p>

In [3]:
# Calcular el porcentaje de valores faltantes en cada atributo
porcentaje_nulos = df.isnull().sum() / len(df) * 100
print("--- Porcentaje de valores faltantes por columna ---")
print(porcentaje_nulos)

print("\n")

# Identificar qué columnas son numéricas y cuáles categóricas
print("--- Tipos de datos por columna ---")
print(df.dtypes)

--- Porcentaje de valores faltantes por columna ---
FlightNumber       0.000000
Date               0.000000
BoosterVersion     0.000000
PayloadMass        0.000000
Orbit              0.000000
LaunchSite         0.000000
Outcome            0.000000
Flights            0.000000
GridFins           0.000000
Reused             0.000000
Legs               0.000000
LandingPad        28.888889
Block              0.000000
ReusedCount        0.000000
Serial             0.000000
Longitude          0.000000
Latitude           0.000000
dtype: float64


--- Tipos de datos por columna ---
FlightNumber        int64
Date                  str
BoosterVersion        str
PayloadMass       float64
Orbit                 str
LaunchSite            str
Outcome               str
Flights             int64
GridFins             bool
Reused               bool
Legs                 bool
LandingPad            str
Block             float64
ReusedCount         int64
Serial                str
Longitude         float64
Lati

<div style="background-color: #f4f6f7; padding: 15px; border-left: 5px solid #24a148;">
    <h2 style="color: #24a148; margin-top: 0;">Interpretación de los Resultados</h2>
    <p><strong>Qué significa esto:</strong> Los resultados confirman que la única variable con datos faltantes es <code>LandingPad</code>, con aproximadamente un 28.89% de valores nulos. Esto tiene sentido operativo, ya que no todos los cohetes de SpaceX intentan aterrizar o tienen una plataforma asignada. En cuanto a los tipos de datos, la distribución es correcta: características como <code>LaunchSite</code> son de tipo texto (<code>object</code>/<code>str</code>), los indicadores como <code>GridFins</code> son booleanos (<code>bool</code>) y variables como <code>PayloadMass</code> son numéricas (<code>float64</code>)

<div style="background-color: #f4f6f7; padding: 15px; border-left: 5px solid #0f62fe;">
    <h2 style="color: #0f62fe; margin-top: 0;">TAREA 1: Calcular el número de lanzamientos por sitio</h2>
    <p><strong>Qué hacemos:</strong> Utilizamos el método <code>value_counts()</code> sobre la columna <code>LaunchSite</code>. Esto nos permitirá contar la frecuencia de cada valor único en esa columna, revelando cuántos lanzamientos se realizaron desde cada instalación de SpaceX (como CCAFS SLC 40, VAFB SLC 4E y KSC LC 39A).</p>
    <hr style="border: 1px solid #d1d5db;">
    <p><strong>Qué observar:</strong> La celda imprimirá una serie con el nombre de cada sitio de lanzamiento y la cantidad total de registros asociados a él.</p>
</div>

In [4]:
# Calcular el número de lanzamientos para cada sitio usando value_counts()
conteo_lanzamientos = df['LaunchSite'].value_counts()

print("--- Número de lanzamientos por sitio ---")
print(conteo_lanzamientos)

--- Número de lanzamientos por sitio ---
LaunchSite
CCAFS SLC 40    55
KSC LC 39A      22
VAFB SLC 4E     13
Name: count, dtype: int64


<div style="background-color: #f4f6f7; padding: 15px; border-left: 5px solid #24a148;">
    <h2 style="color: #24a148; margin-top: 0;">Interpretación de los Resultados</h2>
    <p><strong>Qué significa esto:</strong> El conteo confirma que el sitio de lanzamiento más utilizado por SpaceX en este conjunto de datos es CCAFS SLC 40 con 55 despegues, seguido por KSC LC 39A con 22 y VAFB SLC 4E con 13. Esta distribución es un dato clave para el análisis exploratorio, ya que muestra una clara preferencia operativa por el complejo de Cabo Cañaveral.</p>

<div style="background-color: #f4f6f7; padding: 15px; border-left: 5px solid #0f62fe;">
    <h2 style="color: #0f62fe; margin-top: 0;">Contexto Teórico: Tipos de Órbitas Comunes</h2>
    <p><strong>Qué hacemos:</strong> Traducimos y revisamos las definiciones de las distintas órbitas objetivo para los lanzamientos de SpaceX. Comprender estas categorías es crucial para interpretar los datos de la columna <code>Orbit</code> en los próximos análisis exploratorios o modelos predictivos.</p>
    <ul>
        <li><strong>LEO (Órbita Terrestre Baja):</strong> Órbita centrada en la Tierra con una altitud de 2.000 km o menos. La mayoría de los objetos hechos por el hombre están en LEO.</li>
        <li><strong>VLEO (Órbita Terrestre Muy Baja):</strong> Órbitas con una altitud media inferior a 450 km. Proporcionan beneficios para las naves de observación de la Tierra al estar más cerca.</li>
        <li><strong>GTO (Órbita de Transferencia Geoestacionaria):</strong> Órbita elíptica utilizada para transferir satélites desde LEO hasta la órbita geoestacionaria (GEO). El perigeo es muy bajo y el apogeo alcanza unos 35.786 km.</li>
        <li><strong>SSO o SO (Órbita Heliosíncrona):</strong> Órbita casi polar donde el satélite pasa sobre cualquier punto de la superficie del planeta a la misma hora solar local media.</li>
        <li><strong>ES-L1:</strong> En los puntos de Lagrange, las fuerzas gravitacionales de dos grandes cuerpos se cancelan. L1 es uno de esos puntos entre el Sol y la Tierra.</li>
        <li><strong>HEO (Órbita Altamente Elíptica):</strong> Órbita elíptica con alta excentricidad, generalmente alrededor de la Tierra. (Nota: en la literatura de IBM a veces también se usa para referirse a órbitas por encima de la altitud geosíncrona).</li>
        <li><strong>ISS:</strong> Estación Espacial Internacional. Un satélite artificial habitable modular en órbita terrestre baja (LEO).</li>
        <li><strong>MEO (Órbita Terrestre Media):</strong> Órbitas geocéntricas con altitudes entre 2.000 km y justo por debajo de la órbita geosíncrona. Tienen comúnmente un período orbital de 12 horas.</li>
        <li><strong>GEO (Órbita Geoestacionaria):</strong> Órbita circular geosíncrona a 35.786 km sobre el ecuador de la Tierra y que sigue la dirección de rotación de esta.</li>
        <li><strong>PO (Órbita Polar):</strong> Un tipo de órbita en la que el satélite pasa por encima o casi por encima de ambos polos del cuerpo orbitado.</li>
    </ul>

<div style="background-color: #f4f6f7; padding: 15px; border-left: 5px solid #0f62fe;">
    <h2 style="color: #0f62fe; margin-top: 0;">TAREA 2: Calcular el número y ocurrencia de cada órbita</h2>
    <p><strong>Qué hacemos:</strong> Determinamos la frecuencia de cada tipo de órbita en la columna <code>Orbit</code> mediante el método <code>value_counts()</code>. Aplicamos un filtro previo al DataFrame para excluir los registros correspondientes a <code>GTO</code>, cumpliendo estrictamente con la indicación del laboratorio de omitir esta órbita de transferencia.</p>

In [5]:
# Filtrar la órbita GTO y calcular las ocurrencias del resto de órbitas
conteo_orbitas = df[df['Orbit'] != 'GTO']['Orbit'].value_counts()

print("--- Número de lanzamientos por órbita (excluyendo GTO) ---")
print(conteo_orbitas)

--- Número de lanzamientos por órbita (excluyendo GTO) ---
Orbit
ISS      21
VLEO     14
PO        9
LEO       7
SSO       5
MEO       3
ES-L1     1
HEO       1
SO        1
GEO       1
Name: count, dtype: int64


<div style="background-color: #f4f6f7; padding: 15px; border-left: 5px solid #24a148;">
    <h2 style="color: #24a148; margin-top: 0;">Interpretación de los Resultados</h2>
    <p><strong>Qué significa esto:</strong> El conteo confirma que, excluyendo la órbita de transferencia GTO, el destino más frecuente de SpaceX es la Estación Espacial Internacional (<strong>ISS</strong>) con 21 misiones, seguida por la órbita terrestre muy baja (<strong>VLEO</strong>) con 14. Esto destaca la importancia de las misiones de reabastecimiento tripuladas/no tripuladas y el despliegue de constelaciones de satélites cercanos a la Tierra dentro del conjunto de datos.</p>

<div style="background-color: #f4f6f7; padding: 15px; border-left: 5px solid #0f62fe;">
    <h2 style="color: #0f62fe; margin-top: 0;">TAREA 3: Calcular el número y ocurrencia de los resultados de las misiones</h2>
    <p><strong>Qué hacemos:</strong> Aplicamos el método <code>value_counts()</code> sobre la columna <code>Outcome</code> para contabilizar los diferentes resultados de los aterrizajes y asignamos esta serie a la variable <code>landing_outcomes</code>.</p>

In [6]:
# Calcular los resultados de los aterrizajes usando value_counts() en la columna Outcome
landing_outcomes = df['Outcome'].value_counts()

print("--- Resultados de aterrizaje (Outcome) ---")
print(landing_outcomes)

--- Resultados de aterrizaje (Outcome) ---
Outcome
True ASDS      41
None None      19
True RTLS      14
False ASDS      6
True Ocean      5
False Ocean     2
None ASDS       2
False RTLS      1
Name: count, dtype: int64


<div style="background-color: #f4f6f7; padding: 15px; border-left: 5px solid #0f62fe;">
    <h2 style="color: #0f62fe; margin-top: 0;">Contexto Teórico: Resultados de Aterrizaje</h2>
    <p><strong>Qué hacemos:</strong> Revisamos la nomenclatura de los resultados de aterrizaje (columna <code>Outcome</code>) para comprender el éxito o fracaso de las misiones de SpaceX. Esta clasificación será clave para crear nuestra variable objetivo (target) más adelante.</p>
    <ul>
        <li><strong>True Ocean / False Ocean:</strong> Aterrizaje exitoso (True) o fallido (False) en una región específica del océano.</li>
        <li><strong>True RTLS / False RTLS:</strong> Aterrizaje exitoso (True) o fallido (False) en una plataforma terrestre (Return to Launch Site).</li>
        <li><strong>True ASDS / False ASDS:</strong> Aterrizaje exitoso (True) o fallido (False) en un barco dron (Autonomous Spaceport Drone Ship).</li>
        <li><strong>None ASDS / None None:</strong> Representan un fallo al aterrizar (generalmente porque no hubo intento o la recuperación no fue planificada).</li>
   

<div style="background-color: #f4f6f7; padding: 15px; border-left: 5px solid #0f62fe;">
    <h2 style="color: #0f62fe; margin-top: 0;">Clasificar Resultados de Aterrizaje (Bad Outcomes)</h2>
    <p><strong>Qué hacemos:</strong> Iteramos sobre los diferentes resultados de aterrizaje de la variable <code>landing_outcomes</code> para identificar la posición de cada uno. Luego, extraemos un conjunto (<code>set</code>) con los resultados que representan un fracaso utilizando los índices proporcionados por el laboratorio de IBM. Para garantizar una ejecución segura en pandas 3.0.3, utilizamos el atributo <code>.index</code> en lugar de <code>.keys()</code>.</p>

In [7]:
# Mostrar cada resultado con su índice correspondiente
for i, outcome in enumerate(landing_outcomes.index):
    print(i, outcome)

# Agrupar los resultados fallidos (bad_outcomes) basados en sus índices
bad_outcomes = set(landing_outcomes.index[[1, 3, 5, 6, 7]])

print("\n--- Resultados categorizados como fallidos (bad_outcomes) ---")
print(bad_outcomes)

0 True ASDS
1 None None
2 True RTLS
3 False ASDS
4 True Ocean
5 False Ocean
6 None ASDS
7 False RTLS

--- Resultados categorizados como fallidos (bad_outcomes) ---
{'False Ocean', 'False ASDS', 'False RTLS', 'None ASDS', 'None None'}


<div style="background-color: #f4f6f7; padding: 15px; border-left: 5px solid #0f62fe;">
    <h2 style="color: #0f62fe; margin-top: 0;">TAREA 4: Crear una etiqueta de resultado de aterrizaje</h2>
    <p><strong>Qué hacemos:</strong> Convertimos los resultados de aterrizaje en una variable de clasificación binaria. Iteramos sobre la columna <code>Outcome</code> para crear la lista <code>landing_class</code>: si el valor pertenece al conjunto <code>bad_outcomes</code>, le asignamos 0 (fracaso); de lo contrario, le asignamos 1 (éxito). Este paso es fundamental para definir la variable objetivo (target) que usarán los algoritmos predictivos.</p> 

In [8]:
# Crear la variable de clasificación binaria
landing_class = [0 if outcome in bad_outcomes else 1 for outcome in df['Outcome']]

# Visualizar los primeros 10 elementos de la lista
print("--- Primeros 10 valores de landing_class ---")
print(landing_class[:10])

--- Primeros 10 valores de landing_class ---
[0, 0, 0, 0, 0, 0, 1, 1, 0, 0]


<div style="background-color: #f4f6f7; padding: 15px; border-left: 5px solid #0f62fe;">
    <h2 style="color: #0f62fe; margin-top: 0;">Asignar la Variable Objetivo y Exportar los Datos</h2>
    <p><strong>Qué hacemos:</strong> Integramos la lista <code>landing_class</code> que creamos en el paso anterior como una nueva columna llamada <code>Class</code> en nuestro DataFrame. Luego, calculamos la tasa media de éxito de los aterrizajes (utilizando el método <code>mean()</code>) y, finalmente, exportamos el conjunto de datos limpio y preparado a un archivo CSV (<code>dataset_part_2.csv</code>) que será la base para el siguiente laboratorio.</p>

In [9]:
# Asignar la lista de resultados a la nueva columna 'Class'
df['Class'] = landing_class

# Verificar las primeras 8 filas de la columna Class
print("--- Primeras 8 filas de la columna Class ---")
print(df[['Class']].head(8))

# Verificar las primeras 5 filas del DataFrame completo
print("\n--- Primeras 5 filas del DataFrame actualizado ---")
print(df.head(5))

# Calcular la tasa de éxito global
tasa_exito = df["Class"].mean()
print("\n--- Tasa de éxito global (Media de Class) ---")
print(tasa_exito)

# Exportar el dataset para el siguiente laboratorio
df.to_csv("dataset_part_2.csv", index=False)
print("\nArchivo 'dataset_part_2.csv' exportado exitosamente.")

--- Primeras 8 filas de la columna Class ---
   Class
0      0
1      0
2      0
3      0
4      0
5      0
6      1
7      1

--- Primeras 5 filas del DataFrame actualizado ---
   FlightNumber        Date BoosterVersion  PayloadMass Orbit    LaunchSite  \
0             1  2010-06-04       Falcon 9  6104.959412   LEO  CCAFS SLC 40   
1             2  2012-05-22       Falcon 9   525.000000   LEO  CCAFS SLC 40   
2             3  2013-03-01       Falcon 9   677.000000   ISS  CCAFS SLC 40   
3             4  2013-09-29       Falcon 9   500.000000    PO   VAFB SLC 4E   
4             5  2013-12-03       Falcon 9  3170.000000   GTO  CCAFS SLC 40   

       Outcome  Flights  GridFins  Reused   Legs LandingPad  Block  \
0    None None        1     False   False  False        NaN    1.0   
1    None None        1     False   False  False        NaN    1.0   
2    None None        1     False   False  False        NaN    1.0   
3  False Ocean        1     False   False  False        NaN    1.0 

<div style="background-color: #f4f6f7; padding: 15px; border-left: 5px solid #24a148;">
    <h2 style="color: #24a148; margin-top: 0;">Interpretación de los Resultados</h2>
    <p><strong>Qué significa esto:</strong> La integración de la variable objetivo fue exitosa. Al calcular la media de la columna <code>Class</code>, confirmamos que la tasa histórica de éxito de los aterrizajes en este conjunto de datos es aproximadamente del 66.67%. Además, has exportado el archivo <code>dataset_part_2.csv</code>, lo que concluye esta sección de preparación de datos; tu dataset ahora está etiquetado y estructurado de forma consistente para los futuros laboratorios de visualización y modelado predictivo.

<div style="background-color: #f4f6f7; padding: 15px; border-left: 5px solid #0f62fe;">
    <h2 style="color: #0f62fe; margin-top: 0;">Cuestionario de Verificación del Laboratorio</h2>
    <p><strong>Qué hacemos:</strong> Resolvemos las preguntas de verificación del laboratorio basándonos en las tareas que acabamos de completar paso a paso en este notebook.</p>
    <ul>
        <li><strong>1. Did you calculate the number of launches on each site?</strong> Sí (Yes). Lo hicimos en la Tarea 1 usando el método <code>value_counts()</code> sobre la columna <code>LaunchSite</code>.</li>
        <li><strong>2. Did you calculate the number and occurrence of each orbit?</strong> Sí (Yes). Lo completamos en la Tarea 2 excluyendo la órbita de transferencia GTO.</li>
        <li><strong>3. Did you calculate the number and occurrence of mission outcome per orbit type?</strong> Sí (Yes). Lo revisamos en la Tarea 3 al analizar las frecuencias de la columna <code>Outcome</code>.</li>
        <li><strong>4. Did you create a landing outcome label from outcome column?</strong> Sí (Yes). Lo realizamos en la Tarea 4 al mapear los resultados fallidos y exitosos en la nueva columna binaria <code>Class</code>.</li>
    </ul>
    <hr style="border: 1px solid #d1d5db;">
    <p><strong>Qué observar:</strong> Todas las respuestas del cuestionario son afirmativas, lo que confirma que completaste exitosamente todos los requerimientos de limpieza y estructuración de datos de esta sección (Data Wrangling).</p>
</div>

In [10]:
# 1. ¿Cuántos lanzamientos se realizaron desde CCAFS SLC 40?
lanzamientos_ccafs = df[df['LaunchSite'] == 'CCAFS SLC 40'].shape[0]
print(f"1. Lanzamientos desde CCAFS SLC 40: {lanzamientos_ccafs}")

# 2. ¿Cuál fue la tasa de éxito? (Redondeada)
tasa_exito_pct = df['Class'].mean() * 100
print(f"2. Tasa de éxito: {tasa_exito_pct:.0f}%")

# 3. ¿Cuántos lanzamientos fueron a órbita geosíncrona (GEO)?
lanzamientos_geo = df[df['Orbit'] == 'GEO'].shape[0]
print(f"3. Lanzamientos a órbita GEO: {lanzamientos_geo}")

# 4. ¿Cuántos aterrizajes fueron exitosos en un barco dron (True ASDS)?
aterrizajes_drone = df[df['Outcome'] == 'True ASDS'].shape[0]
print(f"4. Aterrizajes exitosos en barco dron (True ASDS): {aterrizajes_drone}")

1. Lanzamientos desde CCAFS SLC 40: 55
2. Tasa de éxito: 67%
3. Lanzamientos a órbita GEO: 1
4. Aterrizajes exitosos en barco dron (True ASDS): 41
